paper_framework_sources/p1898_forward_progress_delegation.md

```diff
@@ -301,76 +311,61 @@ When applied to a `receiver`, if supported, will return a `scheduler` that the `
+A `scheduler` representing that execution context will be made available from the `receiver` passed to `connect` on the `sender`, thus making that `receiver` a `scheduler_provider`.
 
-
-### Wording
+### Wording to modify final otherwise clause
  * Otherwise:
    * Constructs an execution context owned by the calling thread such that tasks enqueued into that context will be processed by the calling thread.
-   * Constructs a `scheduler`, `s` representing that execution context.
-   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::submit(S, r)`.
-   * `execution::get_scheduler(r)` will return `s`.   
-   * Waits on the synchronization primitive to block on completion of `S`.
+   * Constructs a `scheduler`, `sch` representing that execution context.
+   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::connect(s, r)` such that:
+     * `r` satisfies `scheduler_provider`.
+     * `execution::get_scheduler(r)` will return `sch`.
+     * If `set_value` is called on `r`, the call to `sync_wait` will return the passed value.
+     * If `set_error` is called on `r`, the call to `sync_wait` will throw the passed value as an exception.
+     * If `set_done` is called on `r`, the call to `sync_wait` will throw a cancellation exception.
+   * Waits on the synchronization primitive to block on completion of `s`, driving the execution context as necessary until `set_value`, `set_error` or `set_done` is called on `r` .
 
 The thread that invokes `sync_wait` will block with forward progress delegation on completion of the work where supported by the implementation of `s`.
 
-
-## execution::on
+## execution::on modifications
 ### Summary
-Transitions execution from one executor to the context of a scheduler.
-Passes a `scheduler_provider` to the passed `sender`, such that it is valid to apply `on` to a sender that restricts its `submit` operation to `scheduler_provider`s.
-That is that:
-```
-sender1 | on(scheduler1) | bulk_execute(f)...
-```
+In addition to transitioning execution from one execution context to another, `on` also changes the `scheduler` that is propagated upstream through `scheduler_provider`s.
 
-will return a sender that runs in the context of `scheduler1` such that `f` will run on the context of `scheduler1`, potentially customized, but that is not triggered until the completion of `sender1`.
-If `sender1` calls `get_scheduler` on the `receiver` passed by `on` to `sender1`'s `submit` operation, then that shall return `scheduler1`.
+### Wording to modify otherwise clause
+Add the clauses:
+   * If `output_receiver` satisfies `scheduler_provider` then `execution::get_scheduler(r2)` returns the result of `execution::get_scheduler(output_receiver)`.
+   * `r` satisfies `scheduler_provider` and `execution::get_scheduler(r2)` returns sch.
```

Comment:
is this expressing 
if a scheduler_provider is directly specified as in `on(provider)` that the specified scheduler_provider will be used to both schedule work in on() and to respond to get_scheduler for the receiver(s) created by on() 
and 
if no scheduler_provider is directly specified as in `on()` that the consuming receiver must also be a scheduler_provider and that this scheduler provider will be used to both schedule work in on() and to respond to get_scheduler for the receiver(s) created by on()?

Reply:
It's confusing because the wording has a typo.

What this is saying is:
 * R2 that is submitted to sch will be a scheduler_provider returning the result of get_scheduler(output_receiver) if output_receiver is a scheduler_provider. This allows sch to delegate downstream.
 * R is a scheduler_provider returning sch, because that's what on is injecting into the pipeline.